### ISSUE TITLE
Respect "Commas first" rule

### ISSUE THREAD

**Issue by garex** [feross/standard#464]
@see https://docs.npmjs.com/misc/coding-style#comma-first

```
var magicWords = [ 'abracadabra'
                 , 'gesundheit'
                 , 'ventrilo'
                 ]
  , spells = { 'fireball' : function () { setOnFire() }
             , 'water' : function () { putOut() }
             }
  , a = 1
  , b = 'abc'
  , etc
  , somethingElse
```

**Comment by yoshuawuyts** [comment]
Thanks for opening this issue. The reply here is similar as on #400

> Sorry, we can't even consider this change because it would break 100% of the packages that use standard today.

> If you're using standard, you can't even make this error, because it's invalid JavaScript, so you'll get a parse error when you run the tool, and probably the first time you try to run your app.

I would be against loosening this rule too, as I find comma-first particularly hard to read. Perhaps others would disagree, but in this case we've _picked something and gone with it_. If this is a deal breaker, then perhaps `standard` is not for you. I hope you understand. Thanks!

**Comment by garex** [comment]
I also think that this is not too good to read. But. There is one but:

**More clear diffs.**

When default comma-last rule used, we have this in diffs:

```
otherItem,
- existingItem
+ existingItem,
+ newItem
```

But in comma-first case this will be more clear and just one line changes:

```
, otherItem
, existingItem
+ newItem
```

And we clearly see, that `newItem` was added. If you are respecting diff-readers, you should remeber this case.

**Comment by LinusU** [comment]
@garex You have only moved the problem to the first line instead of the last though. Another option would be to allow trailing comma, it doesn't look nearly as bad and handle diffs very nice as well...

**Comment by garex** [comment]
@LinusU, trailing comma is a syntax error for objects and arrays. See #400

**Comment by LinusU** [comment]
Now it isn't. I think that what @feross means is that the error of _missing_ a comma will be caught by standard.

The only place it isn't allowed is in arguments to a function, but there is a proposal up to start allowing that...

**Comment by garex** [comment]
> Now it isn't.

@LinusU Are you sure? In which popular browsers this is not error?

**Comment by LinusU** [comment]
```javascript
var arr = [
  1,
  2,
]

var obj = {
  a: 1,
  b: 2,
}
```

This works in Node.js, Safari and Chrome; and probably all other browsers/runtimes as well.

**Comment by garex** [comment]
@LinusU see these nice answers: http://stackoverflow.com/questions/7246618/trailing-commas-in-javascript

**TL;DR**

If you are dont' care about compatibility before js5-browsers, then trailing commas are ok for you.
